src/main/java/com/spotify/confidence/ConfidenceValue.java

```diff
@@ -0,0 +1,547 @@
+  }
+
+  public static ConfidenceValue of(Instant value) {
+    return new Timestamp(value);
+  }
+
+  public static ConfidenceValue of(LocalDate date) {
+    return new Date(date);
+  }
+
+  public static ConfidenceValue of(String value) {
+    return new StringValue(value);
+  }
+
+  public static ConfidenceValue of(boolean value) {
+    return new BooleanValue(value);
+  }
+
+  public static ConfidenceValue.List ofStrings(java.util.List<String> values) {
+    return new List(values.stream().map(ConfidenceValue::of).collect(Collectors.toList()));
+  }
+
+  public static ConfidenceValue.List ofBooleans(java.util.List<Boolean> values) {
+    return new List(values.stream().map(ConfidenceValue::of).collect(Collectors.toList()));
+  }
+
+  public static ConfidenceValue.List ofIntegers(java.util.List<java.lang.Integer> values) {
+    return new List(values.stream().map(ConfidenceValue::of).collect(Collectors.toList()));
+  }
+
+  public static ConfidenceValue.List ofDoubles(java.util.List<java.lang.Double> values) {
+    return new List(values.stream().map(ConfidenceValue::of).collect(Collectors.toList()));
+  }
+
+  public static ConfidenceValue.List ofTimestamps(java.util.List<Instant> values) {
+    return new List(values.stream().map(ConfidenceValue::of).collect(Collectors.toList()));
+  }
+
+  public static ConfidenceValue.List ofDates(java.util.List<LocalDate> values) {
+    return new List(values.stream().map(ConfidenceValue::of).collect(Collectors.toList()));
+  }
+
+  public static Struct of(Map<String, ConfidenceValue> values) {
+    return new Struct(values);
+  }
+
+  static ConfidenceValue fromProto(com.google.protobuf.Value protoValue) {
+    final com.google.protobuf.Value.KindCase kind = protoValue.getKindCase();
+    switch (kind) {
+      case BOOL_VALUE:
+        return ConfidenceValue.of(protoValue.getBoolValue());
+      case NUMBER_VALUE:
+        return ConfidenceValue.of(protoValue.getNumberValue());
+      case STRING_VALUE:
+        final String stringValue = protoValue.getStringValue();
+        try {
+          return ConfidenceValue.of(Instant.parse(stringValue));
+        } catch (Exception e1) {
+          try {
+            return ConfidenceValue.of(LocalDate.parse(stringValue));
+          } catch (Exception e2) {
+            return ConfidenceValue.of(stringValue);
+          }
+        }
+      case NULL_VALUE:
+        return NULL_VALUE;
+      case STRUCT_VALUE:
+        return Struct.fromProto(protoValue.getStructValue());
+      case LIST_VALUE:
+        final java.util.List<ConfidenceValue> list =
+            protoValue.getListValue().getValuesList().stream()
+                .map(ConfidenceValue::fromProto)
+                .collect(Collectors.toList());
+        return new List(list);
+    }
+    throw new IllegalArgumentException("Unsupported value kind:" + kind);
+  }
+
+  public abstract com.google.protobuf.Value toProto();
+
+  public static class StringValue extends ConfidenceValue {
```

Comment:
This one differs by having the `Value` suffix in it's name. But I think that makes sense. Having these primitive types being public and sharing so many names with standard Java types will be a mess for developers? We should have a common naming scheme though.

Reply:
Would also be solved by only modeling a Primitive..

Reply:
I'll take a stab at this.